modules/javafx.graphics/src/main/java/com/sun/javafx/sg/prism/NGNode.java

```diff
@@ -330,12 +330,12 @@ public void setTransformedBounds(BaseBounds bounds, boolean byTransformChangeOnl
         // cache needs to be regenerated. So we will not invalidate it here.
         if (dirtyBounds.isEmpty()) {
             dirtyBounds = dirtyBounds.deriveWithNewBounds(transformedBounds);
-            dirtyBounds = dirtyBounds.deriveWithUnion(bounds);
         } else {
             // TODO I think this is vestigial from Scenario and will never
             // actually occur in real life... (RT-23956)
```

Comment:
The comments should be removed and updated to explain the scenario when flow enters the else block and I think RT-23956 can be closed with this fix.

Reply:
I updated the comment. I agree, RT-23956 can now be closed.